### ISSUE TITLE
a11y user review

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/john-travoltage#125]
We received this review from an informal user test.

> I use JAWS 16 Professional edition.  The John Travoltage sim works very well, however when you move the hand or foot it announces something like "position 1" which I have no idea what this means. Position 1 in relation to what? I feel that leaving "very close to the door knob" by itself is helpful enough. If any additional positioning information were to be given it should be in degrees for instance "his arm is at a 90 degree angle with hand pointing towards the ceiling". I also noticed it has to count the charges one at a time. For instance if I swipe and add 3 charges instead of saying "total electrons colon 3" it will go "total electrons colon 1, total electrons colon 2, total electrons colon 3" before stopping. Could it just get total as waiting for it to count up to something 10 or higher is a little annoying and time consuming. It could also just read "total electrons 1" instead of "total electrons colon 1" we understand the colon can be inferred and this takes less time but this isn't too big of an issue.

@jobara @jhung do you have any thoughts on these comments?

**Comment by jobara** [comment]
@jessegreenberg in regards to the electron reporting. This part of the code was intended to address that issue https://github.com/phetsims/john-travoltage/blob/master/js/john-travoltage/view/ElectronLayerNode.js#L78-L86 I wonder if the delay is too short for JAWS, although changing it might be problematic for other environments.

**Comment by jhung** [comment]
We discussed using degrees, radians, and other forms of measurement early on. Those measurements only made sense relative an understood "home" position. We could say something like: "45 degrees from the doorknob".

Some reasons why we didn't choose to do it this way was:

1. It introduced math / units which may not be pedagogically important.
2. It may place too much emphasis on the relative position. If everything is relative to the ceiling for example, users may falsely think that the ceiling is important. If everything is relative to the doorknob, then we may lead the user too much to the doorknob.

Maybe this is a question to ask @ariel-phet ?

Also the queuing of the electrons being reported I think depends on the screen reader? Doesn't Voice Over cancel the message if a new one appears?

**Comment by jobara** [comment]
@jhung yes, queuing of messages is likely somewhat AT dependent. Although in this case we are attempting to not fire events as often.

**Comment by terracoda** [comment]
@jobara @jhung, what we are trying for balloons is differentiating what is happening with state information. For example, when a user first picks up charges they get an alert that they picked up charges from the sweater. When they pick up more they get a full update of the the balloon and sweater, then when the pick up more the get a much shorter update that they have picked up more negative charges. Something similar might work for John Travoltage.

Here's
- Foot rubs carpet. A few negative charges picked up from carpet. 
- More rubbing. A few more charges picked up.
- More rubbing. John, has picked up several (or exact number) negative charges.

Not sure if something like this would work? You likely need other details about what the charge are doing as well.

**Comment by terracoda** [comment]
@jobara, @jhung, this one by one counting of the charges also happened on VO with Safari in Mac OS Sierra (MacBook Air), but it didn't happen on VO & Safari with El Capitan (MacBook Pro).

**Comment by jessegreenberg** [comment]
On JAWS 17 with IE and NVDA with Firefox, I only hear an up to date alert for the number of charges once, I don't hear a queue of multiple updates.

**Comment by jessegreenberg** [comment]
@jobara have you considered sending the alert on the keyup event from the range input?  This might only send a single alert when the user has completed an interaction.  NVDA and JAWS seem to allow keyup events from arrow keys while interacting with ```input type=range``` so it might be an alternative to debounce if we decide this is an issue worth investigating.

**Comment by jobara** [comment]
@jessegreenberg no that hasn't been investigated yet. I believe currently it makes use of changes to the model as opposed to checking the direct interaction.

**Comment by jessegreenberg** [comment]
@jobara understood, and I think that that approach would be better in general as it more closely follows the MVC pattern.

**Comment by jessegreenberg** [comment]
Maybe a combination would work.  A public property could store the string that gets updated with the model, and keyup simply gets that field and passes it on to the PDOM.

**Comment by jessegreenberg** [comment]
#147 was related to this, since that issue was closed, closing this one as well.
